### DISCUSSION TITLE
Slider easings in the osu!editor

### DISCUSSION THREAD

**Discussion by pzrc** [ppy/osu#13016] [Ideas]
**Describe the feature:** Sliders currently offer many possibilities regarding velocity and shapes, but one feature that I think would allow for even more functionality is the use of easing functions in sliders. For example, these easings could be used on sliders to highlight sounds or even sections that increase or decrease in volume or pitch, regardless of the length. Obviously, I am not suggesting adding easings like Back, Elastic, or Bounce that either would be unplayable or are quite literally not how sliders in osu! work.

**Proposal designs of the feature:** Something like this could be added in a similar fashion to the sampleset and additions dropdown in the stable editor.
![design_proposal](https://user-images.githubusercontent.com/46831708/53201087-bbbe3380-35e8-11e9-82b1-b86a510ba1bc.PNG)

**Comment by WebFreak001** [comment]
I would like to pick up this idea again and want to suggest an example implementation how this could look like with more control:

I would suggest a "Slider ball path" property for sliders which can be used to adjust how the ball travels along the slider instead of traveling linearly: (design spec required)

![example slider ball path property using a curve editor](https://wfr.moe/f6yJ8l.png)

![slider ball property stopping and starting again](https://wfr.moe/f6yJ1U.png)

This offers fine control over the ball and can achieve really impressive results with very little implementation work.

**As an example I have modified the code to represent the first curve for every slider in the following map**, where it almost feels like the start of the map has been designed for using these sliders:

https://streamable.com/zuwfl

Sliders using this feel a lot more impactful and can flow over very nicely into faster jumps or gradually slow down a fast section, where currently spamming a lot of slider points would be required to achieve such effects. I think adding this wouldn't be too much work (I can make a few PRs for this) and it would allow mappers a lot more possibilities. When more work on the editor is being done, I think it would make sense to include this early.

For gameplay where currently slider ticks exist, (or should exist) they could be spaced according to the slider ball path to offer some blind readability.

**Comment by lemoncove** [comment]
Slider velocities are already difficult to read as it is—for it to be possible to sight-read this kind of thing, slider ticks would have to appear instantly on sliders. (This is something they used to do in stable a long time ago but that has since changed.)

**Comment by WebFreak001** [comment]
of course this feature can be used to make sliders unreadable, however it can also be used as transition between slider velocities (start like usual and then speed up or slow down to a new default velocity) and for a ranked map it will be needed to get mods too. I think as a tool for mappers this would be very useful both to show impactful long notes and to transition between intensity.

In my test implementation the gameplay isn't fully functional yet (following the slider exactly doesn't give you perfect points, you still have to follow it linearly), but it feels already good to follow such sliders slowly accelerating from nothing.

**Comment by peppy** [comment]
This requires actual design and gameplay testing before actually implementing. It's not currently a focus for us.
